### PR TITLE
Fixes #20879 - Use a different oauth for Pulp & Candlepin

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

File: spec/classes/katello_pulp_spec.rb [hunk 1]
```diff
@@ -11,7 +11,8 @@
         let :pre_condition do
           <<-EOS
           class { '::katello':
-            oauth_secret     => 'secret',
+            pulp_oauth_key        => 'key',
```
Comment (ekohl) [inline spec/classes/katello_pulp_spec.rb]: nit: the alignment is off here.

**Comment by sean797** [comment]
@ekohl thanks for taking a look. I'm fairly certain this wont break any upgrades as puppet will update the keys & secrets everywhere and restart services. But I haven't tested an upgrade.

**Comment by ekohl** [comment]
> @ekohl thanks for taking a look. I'm fairly certain this wont break any upgrades as puppet will update the keys & secrets everywhere and restart services. But I haven't tested an upgrade.

Nowadays you're right it should work because the installer ignores unknown answers and users are not expected to override this but it's still a major version bump per semver.

**Comment by sean797** [comment]
Raised https://github.com/Katello/puppet-katello/pull/212/files for the migration

**Comment by sean797** [comment]
Raised https://github.com/Katello/puppet-katello/pull/212 for the migration

**Comment by ekohl** [comment]
Looks like I caused a merge conflict. Other than that I'd like @ehelms to have a look at this and the migration before I merge it.

**Comment by sean797** [comment]
Updated, thanks!

**Comment by ehelms** [comment]
I wonder if we shouldn't just drop the oauth here for Pulp. We do not use it anymore, as we use client certificate for communication. Oauth is also a deprecated auth for Pulp. That would simplify this update.

**Comment by ekohl** [comment]
[split candlepin/pulp oauth](https://trello.com/c/afmNVNIX/513-split-candlepin-pulp-oauth)

**Comment by sean797** [comment]
@ehelms I didn't know that, I'll look into remove Katello's support for it 👍

**Comment by sean797** [comment]
Closed in favor of https://github.com/Katello/puppet-katello/pull/223

**Pull request closed without merging**